selma/src/main/java/fr/xebia/extras/selma/IoC.java

```diff
@@ -20,5 +20,14 @@
  * Created by slemesle on 27/03/15.
  */
 public enum IoC {
-    NO, SPRING, CDI
+    /** Adds no annotation for IoC */
+    NO,
+    /** Adds @Service annotation to the generated mapper */
+    SPRING, 
+    /** Adds @Inject annotation to the generated mapper */
+    CDI, 
+    /** Adds @Inject and @Singleton annotations to the generated mapper */
+    CDI_SINGLETON, 
```

Comment:
I am not sure why we should create 3 different values for CDI.  Why don't you consider `@Singleton` as required for CDI.